# Disables LTO while running link tests for Ruby extensions.
# If LTO is enabled, link tests sometimes fail when using LLD in 32-bit ARM builds due to https://github.com/llvm/llvm-project/issues/108262.

--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -568,7 +568,7 @@ MSG
 
   def try_link0(src, opt="", *opts, &b) # :nodoc:
     exe = CONFTEST+$EXEEXT
-    cmd = link_command("", opt)
+    cmd = link_command('-fno-lto', opt)
     if $universal
       require 'tmpdir'
       Dir.mktmpdir("mkmf_", oldtmpdir = ENV["TMPDIR"]) do |tmpdir|
